core/vm/instructions.go

```diff
@@ -829,8 +829,13 @@ func opSelfdestruct(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext
 	}
 	beneficiary := scope.Stack.pop()
 	balance := interpreter.evm.StateDB.GetBalance(scope.Contract.Address())
-	interpreter.evm.StateDB.AddBalance(beneficiary.Bytes20(), balance)
-	interpreter.evm.StateDB.Suicide(scope.Contract.Address())
+	if interpreter.evm.chainRules.IsEIP4758 {
+		interpreter.evm.StateDB.SubBalance(scope.Contract.Address(), balance)
+		interpreter.evm.StateDB.AddBalance(beneficiary.Bytes20(), balance)
```

Comment:
Instead of reaching in and modifying the `opSelfDestruct`, IMO it's cleaner to define a new operation, e..g `opSendall`,  and then inserting that on a new instructionset which becomes active at the fork block.